provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -2,34 +2,137 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+pub mod helpers;
+
 use std::collections::BTreeMap;
 
-use crate::transform::cldr::cldr_serde;
+use self::helpers::{
+    contains_alphabetic_chars, convert_array_of_strings_to_fraction, remove_whitespace,
+    split_constant_string, transform_fraction_to_constant_value,
+};
+use crate::transform::cldr::{cldr_serde, units::helpers::is_scientific_number};
 use icu_provider::{
     datagen::IterableDataProvider, DataError, DataLocale, DataPayload, DataProvider, DataRequest,
     DataResponse,
 };
-use icu_unitsconversion::provider::{UnitsConstantsV1, UnitsConstantsV1Marker};
-use zerovec::ZeroMap;
+use icu_unitsconversion::provider::{
+    ConstantType, ConstantValue, UnitsConstantsV1, UnitsConstantsV1Marker,
+};
+use zerovec::{ZeroMap, ZeroVec};
 
 impl DataProvider<UnitsConstantsV1Marker> for crate::DatagenProvider {
     fn load(&self, _req: DataRequest) -> Result<DataResponse<UnitsConstantsV1Marker>, DataError> {
         self.check_req::<UnitsConstantsV1Marker>(_req)?;
 
-        let _units_data: &cldr_serde::units::units_constants::Resource = self
+        let units_data: &cldr_serde::units::units_constants::Resource = self
             .cldr()?
             .core()
             .read_and_parse("supplemental/units.json")?;
-        let mut constants_map = BTreeMap::<&str, &str>::new();
+        let constants = &units_data.supplemental.unit_constants.constants;
+
+        let mut constants_map_in_str_form =
+            BTreeMap::<&str, (Vec<String>, Vec<String>, ConstantType)>::new();
+        for (cons_name, cons_value) in constants {
+            let value = remove_whitespace(&cons_value.value);
```

Comment:
`remove_whitespace` does not need to be called before `split_constant_string`

Reply:
done